### PR TITLE
Automated Updates: suggest property patches to update for Maven

### PR DESCRIPTION
**Pull request by cuixq**
With this PR, the suggester now can generate property patches for Maven, so that the `update` subcommand does not need to handle ecosystem-specific patch information. 

@oliverchang @michaelkedar let me know how you think about this change!

### CODE REVIEW

File: internal/remediation/suggester/maven.go [hunk 1]
```diff
@@ -2,23 +2,124 @@ package suggester
...
+		latest, err := suggestVersion(ctx, client, req, slices.Contains(opts.AvoidMajor, req.Name))
+		if err != nil {
+			return manifest.ManifestPatch{}, fmt.Errorf("suggesting latest version of %s: %w", req.Version, err)
+		}
+		if latest.Version == req.Version {
+			// No need to update
+			continue
+		}
+
+		// TODO: this should go to reporter?
```
Comment (oliverchang) [inline internal/remediation/suggester/maven.go:51]: +1. CC  @another-rex can you give some pointers on how to do this?

**Review by oliverchang — APPROVED** [review]
I assume most of this is the same as the internally reviewed version?

File: internal/remediation/suggester/maven.go (hunk 1 above)
Comment (michaelkedar) [reply internal/remediation/suggester/maven.go:51]: Can this print be moved outside of the `Suggest` function? I don't think this should be printing as a side effect.

File: internal/remediation/suggester/maven.go [hunk 2]
```diff
@@ -2,23 +2,124 @@ package suggester
...
+
+// suggestVersion returns the latest version based on the given Maven requirement version.
 // If there is no newer version available, req will be returned.
 // For a version range requirement,
 //   - the greatest version matching the constraint is assumed when deciding whether the
 //     update is a major update or not.
 //   - if the latest version does not satisfy the constraint, this version is returned;
 //     otherwise, the original version range requirement is returned.
-func (ms *MavenSuggester) Suggest(ctx context.Context, client resolve.Client, req resolve.RequirementVersion, opts SuggestOptions) (resolve.RequirementVersion, error) {
+func suggestVersion(ctx context.Context, client resolve.Client, req resolve.RequirementVersion, noMajorUpdates bool) (resolve.RequirementVersion, error) {
```
Comment (michaelkedar) [inline internal/remediation/suggester/maven.go:122]: nit: `suggestVersion` might get confusing when we add other ecosystems to this suggester submodule.

Maybe make this something like `func suggestMavenVersion()` or `func (ms *MavenSuggester) suggestVersion()`.

File: internal/remediation/suggester/maven.go (hunk 1 above)
Comment (cuixq) [reply internal/remediation/suggester/maven.go]: yes - let me remove this print for now. We can print from the subcommand.

File: internal/remediation/suggester/maven.go (hunk 2 above)
Comment (cuixq) [reply internal/remediation/suggester/maven.go]: done!

**Comment by cuixq** [comment]
yes, most of this is the same as the internally reviewed version - I just move it from the command to suggester so that `update` subcommand is less ecosystem-specific.

**Review by another-rex — APPROVED** [review]

**Pull request merged**